config/initializers/500-admin-dashboard-problems.rb

```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+Rails.application.reloader.to_prepare do
```

Comment:
I _think_ this is right? I basically want to only run this method on boot. The previous way where we had it being called inside the class definition basically breaks local dev, because the reset gets called randomly all the time because the class isn't cached.

Reply:
Sorry I need to clarify why we need to reset the problems when booting the app. From what I understand, we wipe out all the problems whenever the app boots even for the old code. If that is the case, we do not have to store stuff in Redis and can rely on `DistributedCache` instead since there is no need for persistence across deploy.  

Another point to note is that we want to as much as possible try not to make calls to the DB or Redis in initialisers as it makes supporting booting the app with a readonly DB or Redis more difficult.  I know we have code in intialisers that do it but those code gave me a hard time previously when making sure our app can boot in readonly mode.

Reply:
Hmm I am thinking about it and I actually don't need `clear_found_scheduled_check_problems` to be called here. The old code doesn't clear the found problems, it just clears and re-registers all of the checks which I think is fine. If I remove the call to clear then neither Redis nor the DB will be involved here. That is my bad, sorry.

Reply:
@tgxworld does this sound okay then now that the clear is gone?

Reply:
Looking at the API, I think this will still have the same problem previously where things get reset in dev.

> Registers a callback that will run once at application startup and every time the code is reloaded.

Reply:
Hmm you are right :thinking: I actually don't need this anymore, `reset_problem_checks` can just go back into the class definition. It will wipe out checks added by plugins in local development but it won't wipe out any actual problems stored in redis, so I think that is fine for now.